pkg/datapath/linux/devices.go

```diff
@@ -497,10 +499,8 @@ func findK8SNodeIPLink() (netlink.Link, error) {
 // supportL3Dev returns true if the kernel is new enough to support BPF host routing of
 // packets coming from L3 devices using bpf_skb_redirect_peer.
 func supportL3Dev() bool {
-	probesManager := probes.NewProbeManager()
-	if h := probesManager.GetHelpers("sched_cls"); h != nil {
-		_, found := h["bpf_skb_change_head"]
-		return found
+	if err := features.HaveProgramHelper(ebpf.SchedCLS, asm.FnSkbChangeHead); err == nil {
```

Comment:
Replace the whole function with
```suggestion
	return features.HaveProgramHelper(ebpf.SchedCLS, asm.FnSkbChangeHead) == nil
```